v3_usm_test.go

```diff
@@ -228,3 +229,26 @@ func TestIsAuthenticaSHA512(t *testing.T) {
 	require.NoError(t, err, "Authentication check of key failed")
 	require.True(t, authentic, "Packet was not considered to be authentic")
 }
+
+func BenchmarkSingleHash(b *testing.B) {
+	if len(os.Getenv("NOPWDCACHE")) > 0 {
```

Comment:
We should use build tags not ENV vars for controlling tests.

Reply:
As long as your ok with the extra file it is going to add in the top dir, that is fine with me

Reply:
Yes, new files are fine.

Reply:
One alternative, adding `-pwdcache=false` (default true) as a testing argument. We already allow this in the `netsnmp` tests. Execution would look like:

```suggestion
go test -pwdcache=false -bench=.
```

Reply:
Saves the need to create two extra files and is pretty clean. I also now realize i don't neet to call `flag.Parse` in the test so the changes are:

```go
var (
	pwdcache = flag.Bool("pwdcache", true, "enable or disable password caching")
)
```

and 
```go
func BenchmarkSingleHash(b *testing.B) {
	PasswordCaching(*pwdcache)
```

Reply:
I prefer build tags over test flags. What do you think @TimRots?

Reply:
> I prefer build tags over test flags. What do you think @TimRots?

Lets be consistent and stick with build tags as you suggest.

Reply:
Done in the below commit